cmd/crates/stellar-ledger/src/speculos.rs

```diff
@@ -0,0 +1,74 @@
+use std::{collections::HashMap, path::PathBuf};
+use testcontainers::{core::WaitFor, Image, ImageArgs};
+
+const NAME: &str = "docker.io/zondax/builder-zemu";
+const TAG: &str = "speculos-3a3439f6b45eca7f56395673caaf434c202e7005";
+
+#[allow(dead_code)]
+static ENV: &Map = &Map(phf::phf_map! {
+    "BOLOS_SDK"=> "/project/deps/nanos-secure-sdk",
+    "BOLOS_ENV" => "/opt/bolos",
+    "DISPLAY" => "host.docker.internal:0",
+});
+struct Map(phf::Map<&'static str, &'static str>);
+
+#[allow(clippy::implicit_hasher)]
+impl From<&Map> for HashMap<String, String> {
+    fn from(Map(map): &Map) -> Self {
+        map.into_iter()
+            .map(|(a, b)| ((*a).to_string(), (*b).to_string()))
+            .collect()
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct Speculos(HashMap<String, String>, HashMap<String, String>);
+const DEFAULT_APP_PATH: &str = "/project/app/bin";
+impl Speculos {
+    #[allow(dead_code)]
+    pub fn new() -> Self {
+        #[allow(unused_mut)]
+        let apps_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("apps");
+        let mut volumes = HashMap::new();
+        volumes.insert(
+            apps_dir.to_str().unwrap().to_string(),
+            DEFAULT_APP_PATH.to_string(),
+        );
+        Speculos(ENV.into(), volumes)
+    }
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct Args;
+
+impl ImageArgs for Args {
+    fn into_iterator(self) -> Box<dyn Iterator<Item = String>> {
+        let container_elf_path = format!("{DEFAULT_APP_PATH}/stellarNanosApp.elf");
+        let command_string = format!("/home/zondax/speculos/speculos.py --log-level speculos:DEBUG --color JADE_GREEN --display headless -s \"other base behind follow wet put glad muscle unlock sell income october\" -m nanos {container_elf_path}");
```

Comment:
Is the speculos file only used during testing? What's the significance of this hardcoded 12-word passphrase?

Reply:
`speculos.rs` is only used in testing - like I mentioned in the other comment, I'm thinking I should create a `tests` directory or something to put all test-only related files in there to make this more clear.

The hardcoded passphrase is the seed phrase for the account on the emulated device. It probably makes sense to pull this out into a constant.

I can make these changes in a follow-up PR!

Reply:
Ah got it. Yup makes sense.

We've kept tests in other repos in a tests/test module. Rust doesn't require it, but keeping tests in a separate module reduces rebuilds, supposedly, in large rust projects, which is why we use that pattern in other repos. This repo isn't big enough to see any benefit of that though, so not something we need to do. But moving stuff like this only used by tests makes sense.